lib/include/jxl/encode.h

```diff
@@ -573,7 +573,13 @@ JXL_EXPORT void JxlEncoderInitBasicInfo(JxlBasicInfo* info);
  *
  * If the JxlBasicInfo contains information of extra channels beyond an alpha
  * channel, then @ref JxlEncoderSetExtraChannelInfo must be called between
- * JxlEncoderSetBasicInfo and @ref JxlEncoderAddImageFrame.
+ * JxlEncoderSetBasicInfo and @ref JxlEncoderAddImageFrame. In order to indicate
+ * extra channels, the value of `info.num_extra_channels` should be set to the
+ * number of extra channels, also counting the alpha channel if present. If an
+ * alpha channel is indicated by `info.alpha_bits` being non-zero and no other
+ * extra channel is present, then `info.num_extra_channels` should be set to
+ * `1`. However in this case we also allow a value of `0` for
+ * `info.num_extra_channels`.
```

Comment:
It feels a bit wrong to have this behavior where num_extra_channels is allowed to be incorrect in some specific (but common) case.

I think backwards compatibility shouldn't be a huge concern yet for the encode api (we already broke a few things anyway), so I prefer breaking things now and having a clean API (even if it breaks some of the early adopters) rather than already having a wonky API with some 'legacy' idiosyncrasies in it before we reach 1.0.

See also https://github.com/libjxl/libjxl/pull/823#issuecomment-970448469

Reply:
I'm all for a clean API. 
So even when forgetting about backwards compatibility, the question remains if it would be more convenient not to have to set 
`info.num_extra_channels` to 1 in the case that you have an interleaved images with alpha. Do we want to be strict here or not. In any case, I think we should document very clearly what is needed for the encoder API to work.

Reply:
changed to the more strict behavior, so that `num_extra_channels` always has to be correct, i.e. at least 1 if alpha channel is present.